services/github/github-actions-workflow-status.service.js

```diff
@@ -0,0 +1,101 @@
+import Joi from 'joi'
+import { isBuildStatus, renderBuildStatusBadge } from '../build-status.js'
+import { NotFound } from '../index.js'
+import { GithubAuthV3Service } from './github-auth-service.js'
+import { documentation, errorMessagesFor } from './github-helpers.js'
+
+const schema = Joi.object({
+  workflow_runs: Joi.array()
+    .items(
+      Joi.object({
+        conclusion: Joi.alternatives()
+          .try(isBuildStatus, Joi.equal('no status'))
+          .required(),
+      })
+    )
+    .required()
+    .min(0)
+    .max(1),
+}).required()
+
+const queryParamSchema = Joi.object({
+  event: Joi.string(),
+  branch: Joi.string().required(),
+}).required()
+
+const keywords = ['action', 'actions']
+
+export default class GithubActionsWorkflowStatus extends GithubAuthV3Service {
+  static category = 'build'
+
+  static route = {
+    base: 'github/actions/workflow/status',
+    pattern: ':user/:repo/:workflow+',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'GitHub Workflow Status',
+      namedParams: {
+        user: 'actions',
+        repo: 'toolkit',
+        workflow: 'unit-tests.yml',
+      },
+      queryParams: {
+        branch: 'main',
+      },
+      staticPreview: renderBuildStatusBadge({
+        status: 'passing',
+      }),
+      documentation,
+      keywords,
+    },
+    {
+      title: 'GitHub Workflow Status (with event)',
+      namedParams: {
+        user: 'actions',
+        repo: 'toolkit',
+        workflow: 'unit-tests.yml',
+      },
+      queryParams: {
+        event: 'push',
+        branch: 'main',
+      },
+      staticPreview: renderBuildStatusBadge({
+        status: 'passing',
+      }),
+      documentation,
+      keywords,
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'build',
+  }
+
+  async fetch({ user, repo, workflow, branch, event }) {
+    return await this._requestJson({
+      schema,
+      url: `/repos/${user}/${repo}/actions/workflows/${workflow}/runs`,
+      options: {
+        searchParams: {
+          branch,
+          event,
+          page: '1',
+          per_page: '1',
+          exclude_pull_requests: 'true',
```

Comment:
These params are here for performance reasons. There's huge execution time difference between this query with `exclude_pull_requests=true` (fast) vs `exclude_pull_requests=false` (slow)

Reply:
Interesting. Definitely understand the performance angle, but curious if there's any functional impacts? Am I correctly interpreting this as meaning that runs triggered by a pull request would be excluded?

Reply:
Nope. Passing this means the `pull_requests` array in the JSON response is not populated, but we aren't using any values from it so it doesn't really matter here. We can still get the details of a workflow that is triggered by a PR. For example, if I call https://api.github.com/repos/badges/shields/actions/workflows/enforce-dependency-review.yml/runs?branch=frontend-gha&exclude_pull_requests=true (the `enforce-dependency-review.yml` workflow on our repo only runs on PR triggers), we get a response with all the info we need for a badge. Just the `pull_requests` array is empty.